go/stats/opentsdb/backend.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opentsdb
+
+import (
+	"time"
+)
+
+// backend implements stats.PushBackend
+type backend struct {
+	// closed is set to true when Close is called, preventing further use of
+	// the backend and flushing any pending data.
+	closed bool
+	// The prefix is the name of the binary (vtgate, vttablet, etc.) and will be
+	// prepended to all the stats reported.
+	prefix string
+	// Tags that should be included with every data point. If there's a tag name
+	// collision between the common tags and a single data point's tags, the data
+	// point tag will override the common tag.
+	commonTags map[string]string
+	// writer is used to send data points somewhere (file, http, ...).
+	writer writer
```

Comment:
This is new. Factor HTTP-vs.-file specific stats-writing logic into `writer` implementations.